### ISSUE TITLE
Questioning the Experiment Results in Your Paper

### ISSUE THREAD

**Issue by iMarioCorleone** [HKUDS/DCRec#1]
I recently read your paper and found it to be an interesting and informative read. However, I have some concerns regarding the experiment results presented in your paper.

Upon reviewing your experimental methodology and data analysis, I noticed some significant inconsistencies and irregularities in the results. Specifically, in Table 2, the performance of S3 and ICLR is exceptionally poor, while DuoRec's performance is unexpectedly good, which is inconsistent with previous experimental results and my own experimental observations..

Based on my observations, I have reasons to doubt that you did not make a reasonable parameter setting for the baseline and question the validity and accuracy of the experimental results in your paper. In light of this, I kindly request that  you provide the parameters setting or tuning range for the baseline, as well as any additional information that may help clarify the issues I have raised.

**Comment by yuh-yang** [comment]
Hi @iMarioCorleone ,

Thanks for your issue. We appreciate it that you find our work informative, and understand your doubts on baseline performances. Here we try the best to address the questions you raised.

To start with, please note that the S^3-Rec model we re-implemented based on the Recbole release has been tailored to remove SSL for features since we do not include feature contents in the data. We notate this version as S^3-Rec_ISP in the published paper. This is consistent with strategies from followed works, e.g., ICLRec, DuoRec and CL4SRec, which also included S3 as their baselines. An interesting observation during evaluating these baselines is that S3 in these papers all fail to show prior performances, specifically, worse than SASRec in DuoRec and CL4SRec papers, and even worse than GRU4Rec in the ICLR paper. This indicates that the success of S3 reported in its raw paper relies more on the feature SSL part.

And for ICLRec, we have so far no evident observations to support its outperforming capacity against SOTA baselines like DuoRec and CL4SRec. With our released [unified framework](https://github.com/HKUDS/SSL) for SSL recommendation, we performed full-rank evaluation on ML dataset:

<img width="510" alt="截屏2023-04-24 16 47 07" src="https://user-images.githubusercontent.com/43330997/233946304-b951d361-b087-4e3e-8146-6d2f02b9d509.png">

We can see from the results that ICLRec is also far less performed under full-rank settings.

Though we did not go deeper to see why the performance gap is caused for these two methods you mentioned, we are willing to disclose our tuning ranges to make clarification and for further discussion:

S3:
mask_ratio: [0.2, 0.3, 0.4]
mip_weight: [0.2, 0.4, 0.6]
sp_weight: [0.3, 0.5, 0.7]

ICLRec:
temp: [0.6, 0.8, 1.0]
num_clusters: [256, 512, 1024]

**Comment by iMarioCorleone** [comment]
Thank you for your reply, but I am still confused about the results. 

Firstly, S3 is a two-stage method, and you did not mention the training epochs and learning rates for pretraining and finetuning. In theory, reducing the training epochs for pretraining will make the final result of S3 closer to SASRec. Similarly, ICLRec is a multitask method, and the weight of the self-supervised loss is a critical parameter. Setting this weight close to zero will also make ICLRec's performance closer to SASRec. Therefore, the performance of these two methods should not be worse than SASRec. I think the result is still difficult to convince people. 

Secondly, I noticed that **the test_data is used as external_data for model construction (line 332 in run_DCRec.py) and as the validation set for model tuning  (line 340 in run_DCRec.py)** ? ! Is this reasonable?

**Comment by yuh-yang** [comment]
Hi @iMarioCorleone ,

Thanks for your follow-up. We make following further response to address your concern:

1. The way we train S3 is exactly the same as what is present in the Recbole library. That is, we pretrain the model for 200 epochs and select the best one with the interval of each 10 epochs.
2. The seq_cl_weight is set to 0.1 referring to the HP study in the paper, which suggests that 0.1 is an effective value across the datasets they used. Further, as the author did not discuss the impact of intent_cl_weight in the raw paper, which we set it unchanged as 0.3 in the provided codes. From the experiences of model tuning, we think these two values are fair for SSL sequential recommenders and they are not close-to-zero as you may suspect. But we will take your generous advice to tune these two parameters to see if we can enhance the reproduced performances, and include the newest results asap.
3. I think you may have some misunderstanding regarding the external_data. In this step we actually include only history sequences in the test_data to construct new graphs for model inference. As the training-testing strategy here is sliding window, we have to complete those graph connections invisible in the training data. It is reasonable that if you are making predictions for time $t+1$, you make sure your graphs are in the state at time $t$. The graphs I mentioned are only used in the inference function `fast_predict()`(line 387, dcrec.py). And we construct it only using historical sequences (line 47, run_DCRec.py).

Hope these would better clear up the doubts. :)

**Comment by yuh-yang** [comment]
Hey,

I hope you're doing well. Though this issue is closed due to inactiveness, I recently came across a new observation that I think could be valuable for our previous discussion. I hope you don't mind me bringing it up again.

I checked ICLRec's source codes for another series of experiments and just found that they did not do sequence augmentation in their own experiments, which is a common and basic strategy for training a sequential recommender. I think this might be a reason for the performance gap between its reported result ranking and our reproduced one.

If you have further interesting views or results to share, please directly reply.
